### PR TITLE
wysiwyg on csv export

### PR DESCRIPTION
**Pull request by ger-benjamin**
When you export rows in csv from the FeaturesGrid, you may now get only visible columns and keep columns orders.

### CODE REVIEW

File: core/src/script/CGXP/plugins/FeaturesGrid.js [hunk 1]
```diff
@@ -276,8 +276,12 @@ cgxp.plugins.FeaturesGrid = Ext.extend(cgxp.plugins.FeaturesResult, {
      */
     csvEncoding: 'UTF-8',
 
-    /** private: attribute[selectAll]
+    /** api: config[csvWysiwyg]
+     *  ``String``  if true, export only the visibles columns and in
```
Comment (asaunier) [inline core/src/script/CGXP/plugins/FeaturesGrid.js]: visible

File: core/src/script/CGXP/plugins/FeaturesGrid.js [hunk 2]
```diff
@@ -339,26 +347,47 @@ cgxp.plugins.FeaturesGrid = Ext.extend(cgxp.plugins.FeaturesResult, {
...
-                for (var prop in attributes) {
-                    if (attributes.hasOwnProperty(prop)) {
-                        // special IE as it doesn't handle null element as string
-                        if (attributes[prop] !== null) {
-                            properties.push(q + attributes[prop].replace(q, q+q) + q);
-                        } else {
-                            properties.push(q + q);
+
+                //Get header ids
+                if (index==0) {
```
Comment (asaunier) [inline core/src/script/CGXP/plugins/FeaturesGrid.js]: missing spaces around ==

File: core/src/script/CGXP/plugins/FeaturesGrid.js [hunk 3]
```diff
@@ -276,8 +276,12 @@ cgxp.plugins.FeaturesGrid = Ext.extend(cgxp.plugins.FeaturesResult, {
      */
     csvEncoding: 'UTF-8',
 
-    /** private: attribute[selectAll]
```
Comment (ger-benjamin) [inline core/src/script/CGXP/plugins/FeaturesGrid.js]: I don't know why this line exists. I've thought that this is an old thing and have just remove that.

File: core/src/script/CGXP/plugins/FeaturesGrid.js (hunk 2 above)
Comment (asaunier) [inline core/src/script/CGXP/plugins/FeaturesGrid.js]: and missing space after the // commenting signs (ditto below)

File: core/src/script/CGXP/plugins/FeaturesGrid.js [hunk 4]
```diff
@@ -339,26 +347,47 @@ cgxp.plugins.FeaturesGrid = Ext.extend(cgxp.plugins.FeaturesResult, {
...
-                    if (attributes.hasOwnProperty(prop)) {
-                        // special IE as it doesn't handle null element as string
-                        if (attributes[prop] !== null) {
-                            properties.push(q + attributes[prop].replace(q, q+q) + q);
-                        } else {
-                            properties.push(q + q);
+
+                //Get header ids
+                if (index==0) {
+                    //Get only visible column and keep columns order.
```
Comment (asaunier) [inline core/src/script/CGXP/plugins/FeaturesGrid.js]: column => columns :)

File: core/src/script/CGXP/plugins/FeaturesGrid.js [hunk 5]
```diff
@@ -339,26 +347,47 @@ cgxp.plugins.FeaturesGrid = Ext.extend(cgxp.plugins.FeaturesResult, {
...
                         }
+                        csv.push(headerNames.join(this.csvSeparator));
+                    }
+                }
+
+                //Get data for, and order by, previously kept headers
+                var prop;
+                for (i = 0; i < headerIds.length; i++) {
+                    prop = headerIds[i];
+                    // special IE as it doesn't handle null element as string
```
Comment (asaunier) [inline core/src/script/CGXP/plugins/FeaturesGrid.js]: special IE => special code for IE?

File: core/src/script/CGXP/plugins/FeaturesGrid.js (hunk 3 above)
Comment (sbrunner) [inline core/src/script/CGXP/plugins/FeaturesGrid.js]: This line declare that we use a pivate attribute named `selectAll` then she is usefull :-)

File: core/src/script/CGXP/plugins/FeaturesGrid.js [hunk 6]
```diff
@@ -276,8 +276,12 @@ cgxp.plugins.FeaturesGrid = Ext.extend(cgxp.plugins.FeaturesResult, {
      */
     csvEncoding: 'UTF-8',
 
-    /** private: attribute[selectAll]
+    /** api: config[csvWysiwyg]
+     *  ``String``  if true, export only the visible columns and in
+     *  the same order as in the featureGrid.
+     *  Default is false.
      */
+    csvWysiwyg: false,
```
Comment (sbrunner) [inline core/src/script/CGXP/plugins/FeaturesGrid.js]: `wisiwyg` looks a strange naming for me ....
`csvExportVisible` isn't better ?

**Comment by sbrunner** [comment]
This shouldn't be on 1.6 ?

**Comment by sbrunner** [comment]
Just tow small comments, fix them an you can merge :-)

**Comment by ger-benjamin** [comment]
Thanks @asaunier @sbrunner 

You are right, It's preferable to merge on 1.6. I close this pull request and re-open a newer.

**Pull request closed without merging**